CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/attribute/BooleanObjectAttributeDescription.java

```diff
@@ -47,7 +47,7 @@ public BooleanObjectAttributeDescription() {
     @SuppressWarnings("unchecked") // Casts are manually checked
     protected Boolean convertFromObject(final Object object) {
         if (object == null) {
-            return null;
```

Comment:
This was actually an intentional design choice. A boolean object attribute has `null` as its default value, and we take that to essentially mean "no value" as oppose to false.

Reply:
Just wondering what that does as opposed to returning false. In other words where was that being checked for to determine different outcomes for cases of false or null?

Asking because the one time I found where it checked for null was in an inline "if" where if the returned value was null it would return false. (Its one of the four changes made on this pull request, not sure how to link code blocks from the code changes into comments.)

Reply:
To summarise the rule, it basically doesn't like returning null because Boolean can be confused with its primitive counterpart boolean (which can't return null) and so can cause problems with callers code. Based on that, it sounds like we could decide to ignore the rule if we understand the distinction.

On the other hand, lines like the one @sol695510 fixed with "bool == null ? false : bool" looks a bit silly and it would probably be ideal to remove the need for such lines of code

Reply:
Should I just revert the code block mentioned above back to returning a null and keep the others as is?

Reply:
Yes, I think so.

Reply:
To answer your previous question - the Attribute Editor, for example, will display "false" if the value of a boolean object attribute is false and "<No Value>" if it is null. This allows a user to distinguish between, for example, a "false" value retrieved from a database and no value having been set (no data, or no database query performed).

Reply:
> To summarise the rule, it basically doesn't like returning null because Boolean can be confused with its primitive counterpart boolean (which can't return null) and so can cause problems with callers code. Based on that, it sounds like we could decide to ignore the rule if we understand the distinction.
> 
> On the other hand, lines like the one @sol695510 fixed with "bool == null ? false : bool" looks a bit silly and it would probably be ideal to remove the need for such lines of code

> To answer your previous question - the Attribute Editor, for example, will display "false" if the value of a boolean object attribute is false and "" if it is null. This allows a user to distinguish between, for example, a "false" value retrieved from a database and no value having been set (no data, or no database query performed).

Thanks @antares1470 and @cygnus-x-1. I'll make the changes.